### DISCUSSION TITLE
Zed AI: "You must accept the terms of service to use this provider" - Where can I do that?

### DISCUSSION THREAD

**Discussion by manujarvinen** [zed-industries/zed#16607] [Feature Requests]
Hi,

Sorry this might be a useless question, but I've tried long enough to almost give up now.

You have the blog post about Zed AI:
https://zed.dev/blog/zed-ai

I have a Zed account now but in **nowhere** can I find the ability to do what the "Configure your Assistant" panel says, which is:
"You must accept the terms of service to use this provider"

<img width="636" alt="Screenshot 2024-08-21 at 20 41 48" src="https://github.com/user-attachments/assets/3106229a-0acd-4d1e-9a74-dd04628a38f9">

How can this be so hard?
What am I missing?

**Comment by bennetbo** [comment]
Hey @manujarvinen,
sorry for the confusion. Just click on "Open new context", type out a message, send it. Then you should be prompted to accept the terms and conditions.
We should probably provide an option to accept the terms of service in the configuration view.

**Reply by manujarvinen** [reply]
Thank you! Works immediately and nicely!

Here a clarification for others to make it super clear (I hope)

<img width="638" alt="Screenshot 2024-08-21 at 21 15 02" src="https://github.com/user-attachments/assets/adbab477-e131-4539-b8f0-55e8d8b5963b">

**Reply by zeljkoprsa** [reply]
hey @bennetbo it would be great if the ToS is placed inside the config view as well since it doesn't appear in the context window. All I get is this error message:

<img width="414" alt="Screenshot 2024-08-22 at 15 33 51" src="https://github.com/user-attachments/assets/76c33eeb-b437-4c63-8ccc-2321f1b0e3ca">

**Reply by elleszabo** [reply]
> Hey @manujarvinen, sorry for the confusion. Just click on "Open new context", type out a message, send it. Then you should be prompted to accept the terms and conditions. We should probably provide an option to accept the terms of service in the configuration view.

I would like to bump this, it's extremely confusing how to accept the terms and conditions

**Comment by tbenthompson** [comment]
I'm running into this same issue but "Open new context" has no effect:

![2024-08-21 15 33 15](https://github.com/user-attachments/assets/a9610aa4-b8bf-4a5d-8376-b5d81c406950)

Looking at the logs, it appears that the AI features don't work when I'm on a remote machine?

```
Caused by:
    [7] Couldn't connect to server
2024-08-21T21:32:58.50041Z [ERROR] RPC request CreateContext failed: message CreateContext was not handled
2024-08-21T21:32:59.328226Z [ERROR] RPC request CreateContext failed: message CreateContext was not handled
2024-08-21T21:33:17.211195Z [ERROR] RPC request CreateContext failed: message CreateContext was not handled
2024-08-21T21:33:17.883177Z [ERROR] RPC request CreateContext failed: message CreateContext was not handled
2024-08-21T21:33:18.383464Z [ERROR] RPC request CreateContext failed: message CreateContext was not handled
2024-08-21T21:33:18.92619Z [ERROR] RPC request CreateContext failed: message CreateContext was not handled
2024-08-21T21:33:54.843265Z [WARN] request completed with error: failed to connect to the server
2024-08-21T21:33:54.865978Z [ERROR] no worktrees when constructing ProjectLspAdapterDelegate
2024-08-21T21:34:33.272374Z [WARN] request completed with error: failed to connect to the server
```

**Reply by tbenthompson** [reply]
@bennetbo is it correct that this doesn't work over a remote connection?

**Comment by trevorstr** [comment]
I'm running into this problem as well. 

I am not seeing any pop-up asking me to accept the ToS.

I searched the Command Palette for "new context" and don't see any command in there.

How am I supposed to accept the ToS?

<img width="544" alt="Screenshot 2025-04-18 at 9 12 31 AM" src="https://github.com/user-attachments/assets/8c48cea6-9830-4577-8488-e12c4adfaca9" />

**Comment by zboinek** [comment]
Mee too, the same problem
